.github/workflows/promptflow-tracing-e2e-test.yml

```diff
@@ -21,7 +21,17 @@ env:
 
 
 jobs:
+  authorize:
+    environment:
+      # forked prs from pull_request_target will be run in external environment, domain prs will be run in internal environment
+      ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository &&
+      'external' || 'internal' }}
+    runs-on: ubuntu-latest
+    steps:
+      - run: true
   build:
+    needs: authorize
```

Comment:
checkout need to merge to main. or else you are always testing the main branch

Reply:
Added merge to main step.